common/environment/environment.cpp

```diff
@@ -181,27 +181,48 @@ class CLocalEnvironment : implements IConstEnvironment, public CInterface
 
                 keyPairMap[name] = { absPublicKeyPath.str(), absPrivateKeyPath.str() };
             }
-            Owned<IPropertyTreeIterator> clusterIter = p->getElements("EnvSettings/Keys/Cluster");
-            ForEach(*clusterIter)
+
+/* From 7.0.0 until 7.0.6, the <Keys> section of the environment required
+ * the mappings to be defined as "Cluster" instead of "ClusterGroup" - See: HPCC-21192
+ */
+#define BKWRDCOMPAT_CLUSTER_VS_CLUSTERGROUP
+
+            const char *groupKeysPath = "EnvSettings/Keys/ClusterGroup";
+
+#ifdef BKWRDCOMPAT_CLUSTER_VS_CLUSTERGROUP
+            for (unsigned i=0; i<2; i++) // once for std. "ClusterGroup", 2nd time for legacy "Cluster"
             {
-                IPropertyTree &cluster = clusterIter->query();
-                const char *clusterName = cluster.queryProp("@name");
-                if (isEmptyString(clusterName))
+                if (keyClusterMap.size()) // NB: always 0 1st time around.
                 {
-                    WARNLOG("skipping EnvSettings/Keys/Cluster entry with no name");
-                    continue;
+                    WARNLOG("Invalid configuration: mixed 'Keys/ClusterGroup' definitions and legacy 'Keys/Cluster' definitions found, legacy 'Keys/Cluster' definition will be ignored.");
+                    break;
```

Comment:
It is not a big deal - but this will report old keys found even if there are not any there.
This logic could be moved into the ForEach() right before updating the KeyClusterMap[].

Reply:
yeah you're right, not good, I'll change.

Reply:
@mckellyln - have changed so only checks if any in iterator. Some might have not have get as far as actually qualifying for keyClusterMap[] addition, but they shouldn't be there at all.
This involves an extra iter->first() call, that could be avoided if really cared, but we don't here point.

Reply:
looks good